modules/auth/hash/hash_argon2.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
```

Comment:
The file name could be `argon2.go`, the `hash_` is unnecessary since it's already under `hash` package.

Reply:
That was my first approach, but I ran into the error, that `DefaultHasher` was not initiated.

The problem is, that the `DefaultHasher` is created in `hash.go` while the single hashing algos add themselves in their `init` function. So the hash.go has to run first, what is assured if the single algo's name starts with `hash_`. If there is a cleaner way to init the `DefaultHasher`, please give me a hint.